### PR TITLE
Fix versions of python modules not being set in docker images

### LINKED ISSUE #818: Version 0+unknown
The current docker build process leads to all installed libs in the docker containers being version 0+unknown. This is _largely_ a non issue, because the container itself is versioned, but does mean that the spec generated by flowapi will have that as the version as well.

Presumably the best solution is to have build scripts, which build a wheel for the module and copy that rather than the source files as we do currently.

### PR DESCRIPTION
**Pull request by greenape**
Closes #818

### I have:

- [ ] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [ ] Added tests for any new additions
- [ ] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

Uses versioneer's ability to infer a version number from a source tree to set the version numbers for python modules inside docker containers.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,11 +9,19 @@
 FROM jupyter/scipy-notebook
 
 RUN rm -rf /home/$NB_USER/work
-
+ARG SOURCE_VERSION=0+uknown
```
Comment (maxalbert) [inline Dockerfile]:
```suggestion
ARG SOURCE_VERSION=0+unknown
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -547,15 +547,17 @@ jobs:
           path: /home/circleci/project/
       - attach_workspace:
           at: /home/circleci
-      - setup_remote_docker
+      - setup_remote_docker:
+          version: 18.09.3
```
Comment (maxalbert) [inline .circleci/config.yml]: Out of interest, why do we need to pin the version here?

Comment (greenape) [reply .circleci/config.yml]: Uses an ancient version if we don't - specifically, we need a newer engine so we can use env vars in combination with `COPY`

File: flowapi/flowapi/_version.py [hunk 3]
```diff
@@ -382,7 +382,7 @@ def render_pep440_post(pieces):
 
     The ".dev0" means dirty. Note that .dev0 sorts backwards
     (a dirty tree will appear "older" than the corresponding clean one),
-    but you shouldn't be releasing software with -dirty anyway.
+    but you shouldn't be releasing software with -dirty anyways.
```
Comment (maxalbert) [inline flowapi/flowapi/_version.py]: 😄

File: flowauth/Dockerfile [hunk 4]
```diff
@@ -24,6 +24,11 @@ ENV FLASK_APP flowauth
...
+ARG SOURCE_VERSION=0+unknown
+ENV SOURCE_VERSION=${SOURCE_VERSION}
+ENV SOURCE_TREE=FlowKit-${SOURCE_VERSION}
+WORKDIR /${SOURCE_TREE}/flowauth
 COPY backend .
-RUN pip install --no-cache-dir .
\ No newline at end of file
+RUN python setup.py bdist_wheel && pip install dist/*.whl
+WORKDIR /app
+COPY backend/uwsgi.ini .
```
Comment (maxalbert) [inline flowauth/Dockerfile]: What exactly is the purpose of this `uwsgi.ini` file? Should this be included in the distribution instead (e.g. via `MANIFEST.in`)?

Comment (greenape) [reply flowauth/Dockerfile]: It configures uwsgi, which is serving the app in combination with nginx. So is deployment-side rather than distribution side.

**Pull request merged**